src/GraphQL/Utilities/Visitors/Custom/PatternMatchingVisitor.cs

```diff
@@ -0,0 +1,53 @@
+using System.Text.RegularExpressions;
+using GraphQL.Types;
+
+namespace GraphQL.Utilities.Visitors;
+
+/// <summary>
+/// A schema visitor that adds validation for the <see cref="PatternMatchingDirective"/> directive.
+/// Use as follows to enable validation of arguments and input fields:
+/// <code>
+/// services.AddGraphQL(b => b
+///     .AddSchema&lt;MySchema&gt;()
+///     .ConfigureSchema(s =>
+///     {
+///         s.Directives.Register(new <see cref="PatternMatchingDirective"/>());
+///         s.RegisterVisitor(new <see cref="PatternMatchingVisitor"/>());
+///     }));
+///
+/// field.Field(x => x.Email)
+///     .ApplyDirective("pattern", "regex", @".+\@.+\..+");
+/// </code>
+/// </summary>
+public class PatternMatchingVisitor : BaseSchemaNodeVisitor
+{
+    /// <inheritdoc/>
+    public override void VisitInputObjectFieldDefinition(FieldType field, IInputObjectGraphType type, ISchema schema)
+        => field.Validator += GetValidator(field);
```

Comment:
Oh, your new `Validator` feature gets useful. Nice!

Reply:
Yeah, it's getting me very excited.  I've long been wanting to have an easy way to validate values per-argument-definition before they reach the resolver.